### PR TITLE
svu: init at 1.10.2

### PR DESCRIPTION
**Pull request by caarlos0**
###### Description of changes

Add a new package, [svu](https://github.com/caarlos0/svu)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/svu/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    owner = "caarlos0";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "37AT+ygN7u3KfFqr26M9c7aTt15z8m4PBrSd+G5mJcE=";
+  };
+
+  vendorHash = "sha256-+e1oL08KvBSNaRepGR2SBBrEDJaGxl5V9rOBysGEfQs=";
+
+  ldflags =
+    [ "-s" "-w" "-X main.version=${version}" "-X main.builtBy=nixpkgs" ];
```
Comment (figsoda) [inline pkgs/tools/misc/svu/default.nix:17]:
```suggestion
  ldflags = [ "-s" "-w" "-X=main.version=${version}" "-X=main.builtBy=nixpkgs" ];
```

File: pkgs/tools/misc/svu/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
```
Comment (figsoda) [inline pkgs/tools/misc/svu/default.nix:18]: `fetchFromGitHub` uses the archive tarball by default, which doesn't have any git information, alternatively you can set `leaveDotGit = true;` in `fetchFromGitHub` and add git to `nativeCheckInputs` if you think these tests are important

```suggestion

  # test assumes source directory to be a git repository
  postPatch = ''
    rm internal/git/git_test.go
  '';

```

Comment (caarlos0) [reply pkgs/tools/misc/svu/default.nix:18]: does it have `git` though? If so, I can (in a future version) change the tests so they create a test repo (which is probably better anyway)

Comment (figsoda) [reply pkgs/tools/misc/svu/default.nix:22]: > does it have git though?

It does only if you add it

**Review by figsoda — APPROVED** [review]

**Pull request merged**